main/ble2mqtt.c

```diff
@@ -34,7 +35,18 @@ static char *device_name_get(void)
 
     if (!*name)
     {
-        uint8_t *mac = wifi_mac_get();
+        uint8_t *mac;
+        switch (config_network_type_get())
+        {
+        case NETWORK_TYPE_ETH:
+            mac = eth_mac_get();
+            break;
+        case NETWORK_TYPE_WIFI:
+            mac = wifi_mac_get();
+            break;
+        default:
+            mac = wifi_mac_get();
```

Comment:
Please remove the default case. If we add a new network type in the future, I want a compilation error here so it could be addressed.

Reply:
I've removed the default case, however `uint8_t *mac` had to be initialized as null to prevent the compiler from flagging the `sprintf` for having a potentially uninitialized variable. 

https://github.com/shmuelzon/esp32-ble2mqtt/blob/316918dafcd6bb663ae08c3fb445f121a67f3d42/main/ble2mqtt.c#L38-L48